crates/stages/src/stages/bodies.rs

```diff
@@ -109,13 +115,43 @@ impl<DB: Database, D: BodyDownloader> Stage<DB> for BodyStage<D> {
         // Cursors used to write bodies, ommers and transactions
         let tx = provider.tx_ref();
         let mut block_indices_cursor = tx.cursor_write::<tables::BlockBodyIndices>()?;
-        let mut tx_cursor = tx.cursor_write::<tables::Transactions>()?;
         let mut tx_block_cursor = tx.cursor_write::<tables::TransactionBlock>()?;
         let mut ommers_cursor = tx.cursor_write::<tables::BlockOmmers>()?;
         let mut withdrawals_cursor = tx.cursor_write::<tables::BlockWithdrawals>()?;
 
         // Get id for the next tx_num of zero if there are no transactions.
-        let mut next_tx_num = tx_cursor.last()?.map(|(id, _)| id + 1).unwrap_or_default();
+        let mut next_tx_num = tx_block_cursor.last()?.map(|(id, _)| id + 1).unwrap_or_default();
+
+        let snapshot_provider = provider.snapshot_provider().expect("should exist");
```

Comment:
hmm, I don't really like this panic.

I guess this is an expect because unreachable?

Reply:
yeah, stages [should] receive a `DatabaseProvider` with a `SnapshotProvider`